artview/components/plot_points.py

```diff
@@ -303,7 +303,18 @@ def NewPoints(self, variable, strong):
         '''
         # test for None
         if self.Vpoints.value is None:
-            # self.fieldBox.clear()
+            #print(self.layout.itemAt(0).widget())
+            #self.layout.removeWidget(self.layout.itemAt(0).widget())
+            #self.layout.addWidget(QtWidgets.QWidget(), 0)
+            if self.plot_type == 'histogram':
+                self.ax.cla()
+                self.canvas.draw()
+            elif self.plot_type == 'statistics':
+                self.layout.removeWidget(self.statistics)
+                self.statistics.close()
+            elif self.plot_type == 'table':
+                self.layout.removeWidget(self.table)
+                self.table.close()
             return
 
         # Get field names
```

Comment:
I think we may need to add an update method in the `core.common.CreateTable` class to redraw the values in the table. Also, it might be useful to also create a method to replace an individual cell. We could hook this up later to allow table editing.

Check out the following [SO link](http://stackoverflow.com/questions/24371274/qtextedit-how-to-dynamically-update-in-pyqt-without-crashing-gui) for the refreshing text.

Are the issues lies in `changePlotType` method, where it passes on statistics and table?

One more thought about the QTextEdit, could it be tied to the fact we set `setReadOnly` to True? Or does that just allow the user the ability to modify text?